app/scripts/controllers/main.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name bitcoin-price-demo.controller:MainCtrl
+ * @description
+ * # MainCtrl
+ * Controller of the bitcoin-price-demo
+ */
+var myapp = angular.module('bitcoin-price-demo');
+myapp.controller('MainCtrl', function ($scope, $filter, $resource, $log) {
+
+  $scope.currencies = [{code:'',name:'Loading -'}];
+  $scope.topCurrencies = [{code:'AUD'},{code:'USD'},{code:'EUR'},{code:'GBP'},{code:'JPY'},{code:'CAD'},{code:'CNY'}];
+  $scope.selectedCurrencyCode = $scope.topCurrencies[0].code;
+  $scope.btcPrice = '';
+  $scope.graphLoading = true;
+
+  // CurrencyCode Dropdown
+  $scope.getCurrencies = function() {
+
+    var res = $resource('https://bitpay.com/currencies');
+    res.get('', function(bitpay){
+      // Server Returns:
+      // { data: [ { code: "BTC", symbol: "฿", precision: 6, exchangePctFee: 200, payoutEnabled: true, name: "Bitcoin", plural: "Bitcoin", alts: "btc", minimum: 0.000006, payoutFields: [ "bitcoinAddress" ] }, ...
+      $log.log('Currencies: ', bitpay.data[0].code); // debug
+
```

Comment:
I traced your use of the getCurrencies return and you only ever use the code (at present) so this can be greatly simplified (if we are willing to throw all the data except the codes.

Look at the operation provided by the excellent JS utility 'lodash'. For this example I think all this code boils down to

```
currencies = _.pluck(apiResponse.data, 'code')
nonTop = _.without(currencies, topCurrencyCodes)
#lodash is sweet
```

alternatively by making a copy of top currencies that is a dictionary, you can skip the inner loop and replace with dictionary lookup.